### ISSUE TITLE
[indent] Incorrect expected indentation for jsx spread props

### ISSUE THREAD

**Issue by nwoltman** [eslint/eslint#12574]
**Tell us about your environment**

* **ESLint Version:** v6.6.0
* **Node Version:** v13.1.0
* **npm Version:** v6.12.1

**What parser (default, Babel-ESLint, etc.) are you using?**

`babel-eslint@10.0.3`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "root": true,
  "parser": "babel-eslint",
  "parserOptions": {
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "env": {
    "browser": true,
    "node": false
  },
  "rules": {
    "indent": [2, 2, {"SwitchCase": 1}]
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```jsx
import React from 'React';

export default function Button(props) {
  return (
    <button
      type="button"
      {
        ...props // ESLint is (incorrectly) expecting less indentation here
      }
    />
  );
};
```

```bash
eslint ./index.js
```

**What did you expect to happen?**

No errors

**What actually happened? Please include the actual, raw output from ESLint.**

```
  8:1  error  Expected indentation of 6 spaces but found 8  indent
```

**Are you willing to submit a pull request to fix this bug?**
Sure, if none of the maintainers have time to get to this.

**Comment by kaicataldo** [comment]
I have verified this with the [demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IFwiaW5kZW50XCI6IFsyLCAyLCB7XCJTd2l0Y2hDYXNlXCI6IDF9XSAqL1xuXG5pbXBvcnQgUmVhY3QgZnJvbSAnUmVhY3QnO1xuXG5leHBvcnQgZGVmYXVsdCBmdW5jdGlvbiBCdXR0b24ocHJvcHMpIHtcbiAgcmV0dXJuIChcbiAgICA8YnV0dG9uXG4gICAgICB0eXBlPVwiYnV0dG9uXCJcbiAgICAgIHtcbiAgICAgICAgLi4ucHJvcHMgLy8gRVNMaW50IGlzIChpbmNvcnJlY3RseSkgZXhwZWN0aW5nIGxlc3MgaW5kZW50YXRpb24gaGVyZVxuICAgICAgfVxuICAgIC8+XG4gICk7XG59OyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6eyJqc3giOnRydWV9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=). Agreed that this looks like a bug.
